File: Bangun Datar.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

class BangunDatar {
public:
    virtual double hitungLuas() {
        return 0.0;
    }
};

class Persegi : public BangunDatar {
private:
    double sisi;

public:
    Persegi(double s) : sisi(s) {}

    double hitungLuas() {
        return sisi * sisi;
    }
};

class PersegiPanjang : public BangunDatar {
private:
    double panjang;
    double lebar;

public:
    PersegiPanjang(double p, double l) : panjang(p), lebar(l) {}

    double hitungLuas() {
        return panjang * lebar;
    }
};

class Segitiga : public BangunDatar {
private:
    double alas;
    double tinggi;

public:
    Segitiga(double a, double t) : alas(a), tinggi(t) {}

    double hitungLuas() {
        return 0.5 * alas * tinggi;
    }
};

class Lingkaran : public BangunDatar {
private:
    double jariJari;

public:
    Lingkaran(double r) : jariJari(r) {}

    double hitungLuas() {
        return 3.14159265359 * jariJari * jariJari;
    }
};

int main() {
    cout << "Kalkulator Luas Bangun Datar" << endl;
    cout << "Pilih bentuk:" << endl;
    cout << "1. Persegi" << endl;
    cout << "2. Persegi Panjang" << endl;
    cout << "3. Segitiga" << endl;
    cout << "4. Lingkaran" << endl;

    int pilihan;
    cin >> pilihan;

    double panjang, lebar, alas, tinggi, jariJari;

    switch (pilihan) {
        case 1: { // Persegi
            cout << "Masukkan panjang sisi: ";
            cin >> panjang;
            Persegi persegi(panjang);
            cout << "Luas Persegi: " << persegi.hitungLuas() << endl;
            break;
        }
        case 2: { // Persegi Panjang
            cout << "Masukkan panjang: ";
            cin >> panjang;
            cout << "Masukkan lebar: ";
            cin >> lebar;
            PersegiPanjang persegipanjang(panjang, lebar);
            cout << "Luas Persegi Panjang: " << persegipanjang.hitungLuas() << endl;
            break;
        }
        case 3: { // Segitiga
            cout << "Masukkan alas: ";
            cin >> alas;
            cout << "Masukkan tinggi: ";
            cin >> tinggi;
            Segitiga segitiga(alas, tinggi);
            cout << "Luas Segitiga: " << segitiga.hitungLuas() << endl;
            break;
        }
        case 4: { // Lingkaran
            cout << "Masukkan jari-jari: ";
            cin >> jariJari;
            Lingkaran lingkaran(jariJari);
            cout << "Luas Lingkaran: " << lingkaran.hitungLuas() << endl;
            break;
        }
        default:
            cout << "Pilihan tidak valid." << endl;
    }

    return 0;
}

File: Bangun Datar.php
<!DOCTYPE html>
<html>
<head>
    <title>Kalkulator Luas Bangun Datar</title>
</head>
<body>
    <h1>Kalkulator Luas Bangun Datar</h1>
    
    <form action="" method="post">
        <label>Pilih Bentuk:</label>
        <select name="shape">
            <option value="persegi">Persegi</option>
            <option value="persegipanjang">Persegi Panjang</option>
            <option value="segitiga">Segitiga</option>
            <option value="lingkaran">Lingkaran</option>
        </select>
        
        <div id="inputFields">
            <!-- Input fields for user-defined dimensions -->
            <input type="text" name="panjang" placeholder="Panjang">
            <input type="text" name="lebar" placeholder="Lebar">
            <input type="text" name="alas" placeholder="Alas">
            <input type="text" name="tinggi" placeholder="Tinggi">
            <input type="text" name="jariJari" placeholder="Jari-Jari">
        </div>
        
        <input type="submit" value="Hitung Luas">
    </form>
    
    <?php
    if ($_SERVER["REQUEST_METHOD"] == "POST") {
        $shape = $_POST["shape"];
        
        // Definisikan kelas Induk (BangunDatar)
        class BangunDatar {
            public function hitungLuas() {
                return 0;
            }
        }
        
        // Definisikan kelas Anak (Persegi) - Menggunakan inheritance dari BangunDatar
        class Persegi extends BangunDatar {
            private $sisi;

            public function __construct($sisi) {
                $this->sisi = $sisi;
            }

            public function hitungLuas() {
                return $this->sisi * $this->sisi;
            }
        }

        // Definisikan kelas Anak (Persegi Panjang) - Menggunakan inheritance dari BangunDatar
        class PersegiPanjang extends BangunDatar {
            private $panjang;
            private $lebar;

            public function __construct($panjang, $lebar) {
                $this->panjang = $panjang;
                $this->lebar = $lebar;
            }

            public function hitungLuas() {
                return $this->panjang * $this->lebar;
            }
        }

        // Definisikan kelas Anak (Segitiga) - Menggunakan inheritance dari BangunDatar
        class Segitiga extends BangunDatar {
            private $alas;
            private $tinggi;

            public function __construct($alas, $tinggi) {
                $this->alas = $alas;
                $this->tinggi = $tinggi;
            }

            public function hitungLuas() {
                return 0.5 * $this->alas * $this->tinggi;
            }
        }

        // Definisikan kelas Anak (Lingkaran) - Menggunakan inheritance dari BangunDatar
        class Lingkaran extends BangunDatar {
            private $jariJari;

            public function __construct($jariJari) {
                $this->jariJari = $jariJari;
            }

            public function hitungLuas() {
                return 3.14 * $this->jariJari * $this->jariJari;
            }
        }

        // Buat objek berdasarkan pilihan pengguna dan nilai yang dimasukkan
        switch ($shape) {
            case "persegi":
                $panjang = floatval($_POST["panjang"]);
                $shapeObject = new Persegi($panjang);
                break;
            case "persegipanjang":
                $panjang = floatval($_POST["panjang"]);
                $lebar = floatval($_POST["lebar"]);
                $shapeObject = new PersegiPanjang($panjang, $lebar);
                break;
            case "segitiga":
                $alas = floatval($_POST["alas"]);
                $tinggi = floatval($_POST["tinggi"]);
                $shapeObject = new Segitiga($alas, $tinggi);
                break;
            case "lingkaran":
                $jariJari = floatval($_POST["jariJari"]);
                $shapeObject = new Lingkaran($jariJari);
                break;
        }

        // Hitung dan tampilkan luas
        $luas = $shapeObject->hitungLuas();
        echo "<p>Luas $shape adalah $luas</p>";
    }
    ?>
</body>
</html>

